hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -522,15 +537,38 @@ public FileStatus getFileStatus(final Path path) throws IOException {
             eTag);
   }
 
+  /**
+   * @param path The list path.
+   * @return the entries in the path.
+   * */
   public FileStatus[] listStatus(final Path path) throws IOException {
-    LOG.debug("listStatus filesystem: {} path: {}",
+    return listStatus(path, null);
+  }
+
+  /**
+   * @param path Path the list path.
+   * @param startFrom the entry name that list results should start with.
+   *                  Notice that startFrom can be a non-existent entry, then the list request response contains
+   *                  all entries after this non-existent entry in lexical order.
+   * @return the entries in the path in lexical order from "startFrom".
+   * */
+  public FileStatus[] listStatus(final Path path, final String startFrom) throws IOException {
```

Comment:
this doesn't need to be a public API

Reply:
From what I know hadoop-azure jar will be imported by customer code, so I keep it public here. I will add unstable annotation here for now.